addon/validations/factory.js

```diff
@@ -205,28 +208,29 @@ function createCPValidationFor(attribute, validations) {
   var dependentKeys = getCPDependentKeysFor(attribute, validations);
   return computed(...dependentKeys, cycleBreaker(function() {
     var model = get(this, '_model');
-    // var modelErrors = get(model, 'errors');
     var validators = getValidatorsFor(attribute, model);
 
-    var validationResults = validators.map((validator) => {
-      var validationReturnValue = validator.validate(get(model, attribute), validator.processOptions(), model, attribute);
-      return validationReturnValueHandler(attribute, validationReturnValue, model);
-    });
+    var validationResults = validators.map(validator => {
+      let options = validator.processOptions();
+      let debounce = get(options, 'debounce') || 0;
+      let value;
 
-    validationResults = flatten(validationResults);
-    var validationResultsCollection = ValidationResultCollection.create({
-      attribute, content: validationResults
-    });
+      if(debounce > 0) {
+        let cache = getDebouncedValidationsCacheFor(model);
+        // Return a promise and pass the resolve method to the debounce handler
+        value = new Promise(resolve => {
+          cache[attribute] = run.debounce(validator, getValidationResult, validator, options, model, attribute, resolve, debounce, false);
```

Comment:
We likely need to deal with the > 1 debounced per attribute, and also > 1 debounced per attribute per type (2 ajax validators for 1 attribute example).